lib/iris/tests/integration/netcdf/test_delayed_save.py

```diff
@@ -0,0 +1,258 @@
+            assert np.all(~data_mask)
+            assert np.all(~coord_mask)
+
+    @pytest.mark.parametrize(
+        "warning_type", ["WarnMaskedBytes", "WarnFillvalueCollision"]
+    )
+    def test_fill_warnings(self, warning_type, output_path, save_is_delayed):
+        # Test collision warnings for data with fill-value collisions, or for masked
+        # byte data.
+        if warning_type == "WarnFillvalueCollision":
+            make_fv_collide = True
+            make_maskedbytes = False
+            expected_msg = (
+                "contains unmasked data points equal to the fill-value"
+            )
+        else:
+            # warning_type == 'WarnMaskedBytes'
+            make_fv_collide = False
+            make_maskedbytes = True
+            expected_msg = "contains byte data with masked points"
+
+        cube = self.make_testcube(
+            include_lazy_content=True,
+            ensure_fillvalue_collision=make_fv_collide,
+            data_is_maskedbytes=make_maskedbytes,
+        )
+        with warnings.catch_warnings(record=True) as logged_warnings:
+            result = iris.save(cube, output_path, compute=not save_is_delayed)
+
+        if not save_is_delayed:
+            result_warnings = [log.message for log in logged_warnings]
+        else:
+            assert len(logged_warnings) == 0
+            # Complete the operation now
+            # NOTE: warnings should not be *issued* here, instead they are returned.
+            warnings.simplefilter("error")
+            result_warnings = result.compute()
+
+        # Either way, we should now have 2 similar warnings.
+        assert len(result_warnings) == 2
+        assert all(
+            expected_msg in warning.args[0] for warning in result_warnings
+        )
+
+    def test_no_delayed_writes(self, output_path):
+        # Just check that a delayed save returns a usable 'delayed' object, even when
+        # there is no lazy content = no delayed writes to perform.
+        cube = self.make_testcube(include_lazy_content=False)
+        warnings.simplefilter("error")
+        result = iris.save(cube, output_path, compute=False)
+        assert isinstance(result, Delayed)
+        assert result.compute() == []
+
+    _distributed_client = None
+
+    @classmethod
+    @pytest.fixture(params=["ThreadedScheduler", "DistributedScheduler"])
+    def scheduler_type(cls, request):
+        sched_typename = request.param
+        if sched_typename == "ThreadedScheduler":
+            config_name = "threads"
+            if cls._distributed_client is not None:
+                cls._distributed_client.close()
+                cls._distributed_client = None
+        else:
+            config_name = "distributed"
+            if cls._distributed_client is None:
+                cls._distributed_client = distributed.Client()
+
+        with dask.config.set(scheduler=config_name):
+            yield sched_typename
```

Comment:
Running tests locally, I had issues with this one, I think because the distributed scheduler would not be properly cleaned up at the end of the last test. The following changes make sure the distributed scheduler is closed at the end of each test, and it solves the failing tests I got:
```suggestion
    def scheduler_type(cls, request):
        sched_typename = request.param
        if sched_typename == "ThreadedScheduler":
            config_name = "threads"
        else:
            config_name = "distributed"
            if cls._distributed_client is None:
                cls._distributed_client = distributed.Client()

        with dask.config.set(scheduler=config_name):
            yield sched_typename
        if cls._distributed_client is not None:
            cls._distributed_client.close()
            cls._distributed_client = None
```

Reply:
@fnattino thanks,  I see it now !
I really did get this wrong -- we don't want to leave a distributed client in existence after any of these tests, we should always destroy it.
Frankly I'm really surprised this didn't upset other tests within the CI. 
I will fix ...